Tools/jit/jit_definitions.h

```diff
@@ -0,0 +1,25 @@
+#ifndef JIT_DEFINITIONS_H
```

Comment:
I couldn't really find a naming convention precedent here since this is the first header file in `Tools/jit.`  I opted not to include `preserve_none` in the name here since maybe we'll want to include more in the file in the future.

Happy to rename :)

Reply:
How about just `Tools/jit/jit.h`?